engine/src/main/java/org/hibernate/validator/HibernateValidatorConfiguration.java

```diff
@@ -298,4 +298,16 @@
 	 */
 	@Incubating
 	HibernateValidatorConfiguration temporalValidationTolerance(Duration temporalValidationTolerance);
+
+	/**
+	 * Allows to set a payload which will be passed to the constraint validators.
+	 *
+	 * @param constraintValidatorPayload the payload passed to constraint validators
+	 *
+	 * @return {@code this} following the chaining method pattern
+	 *
+	 * @since 6.0.8
+	 */
+	@Incubating
+	HibernateValidatorConfiguration constraintValidatorPayload(Object constraintValidatorPayload);
```

Comment:
How comes this is exposed at the `Configuration` level? IIRC the originally discussed use case, this feature was more indented for a specific context (e.g. the current user's locale) which typically will differ between multiple requests. While I probably can get behind supporting this on a per-validator basis, I don't think exposing this here is very useful. A VF should be constructed once and kept around, you definitely don't want to create a new one for each request.

Reply:
With my PR you don't have to create a new VF for each request, you can just use `vf.unwrap( HibernateValidatorFactory.class ).usingContext().constraintValidatorPayload(...)`. Exposing the payload at `Configuration` level was never my original intention, however during the work on this pull request @gsmet thought we should do that as well, probably to define a "default" payload.

Reply:
This was added at my specific request. I don't think it's that absurd to imagine someone needing to pass information at the VF level to the validators (might be info about the environment for instance, staging/production, with validation rules differing depending on the environment).

And it was also because I wanted the symmetry with the other features.

I think it's better with it than without.

Reply:
I have to agree with @gsmet - it's better with it than without.

Reply:
My concern is that this this may be people lead to do the wrong thing and provide volatile contextual information to the VF instead of getting a (rather cheap) validator instance with such information.

Really that's the purpose of `VF#usingContext()`, I don't think adding another means of passing in "context" is a very good idea. About that environment use case, you'd just pass that to the `Validator` and keep that around. I don't see what's gained by exposing this at the VF level?

Reply:
Why are you supposing that the information is volatile? My environment example is not about volatile information.

I would say it makes as much sense as any other configuration knobs we have at the VF level: we want it to be the default value for all Validator creations.

`usingContext()` is exclusively used to override VF configuration for a given Validator creation. That's how it's documented and coded.

What you're asking is special casing this one, not the other way around.

Reply:
Hmmm... specific to my use case maybe we could run different instances of an application - each instance for a different country. Each country should use it's own locale - no matter what the user locale in the request is. E.g. app instances that server `.de` domains would configure the payload to be german, `.com` instances would be configured to be english, etc.
Just an idea of how that **could** be used (even if that may be not a real good example, just came into my mind right now)

Reply:
@gunnarmorling @mkurz Yes, that's basically what I had in mind.

Or a different behavior in staging for instance because staging doesn't have access to some component required for validation (external service only available for production for instance). I know you shouldn't do that but I also know from experience it happens.